lib/metrics.py

```diff
@@ -95,15 +95,16 @@ def run(self):
         while True:
             stats = {}
             try:
-                if buildpackutil.i_am_primary_instance():
-                    stats = self._inject_database_stats(stats)
-                    stats = self._inject_storage_stats(stats)
-                    stats = self._inject_health(stats)
                 try:
                     stats = self._inject_m2ee_stats(stats)
                 except Exception:
                     logger.debug("Unable to get metrics from runtime")
 
+                if buildpackutil.i_am_primary_instance() and buildpackutil.is_paid_app():
```

Comment:
Instead of gating this behind an `if` statement, I think that it's neater to subclass `MetricsEmitterThread` as a `FreeAppsEmitterThread`, and override the `run` method to call only the sessions stuff.

You can also put the logic to get specific metrics in their own internal `_get_free_apps_session_data` or whatever method.

Reply:
done. Changes in https://github.com/mendix/cf-mendix-buildpack/pull/262